### PR TITLE
Add Video Activities to display and play different channel's videos. 

### LINKED ISSUE #119: Add youtube feed from IITP channels in the app
**Is your feature request related to a problem? Please describe.**
We need to add a section in the project where the user can see all the youtube videos posted by the various IIT Patna channels. Once we have the basic implementation ready, we can also allow them to customize which channel's content they want in their feed.

### PR DESCRIPTION
**Pull request by sheetal-g9**
Fixes #119 

Changes: Add Activities to display and play video of different iitp channels.

### CODE REVIEW

File: app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/ChannelSelectActivity.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        DataBindingUtil.setContentView(this, R.layout.activity_channel_select);
+
+    }
+
+    public void onChannelClick(View view) {
+        String id = getString(getResources().getIdentifier(getResources().getResourceEntryName(view.getId()), "string", getPackageName()));
+        Toast.makeText(this, id, Toast.LENGTH_SHORT).show();
```
Comment (dr0pdb) [inline app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/ChannelSelectActivity.java]: Define a showToast method in the `ViewUtils` file and use that.
File : https://github.com/Njack-IITP/IITP-Connect/blob/development/app/src/main/java/com/iitp/njack/iitp_connect/ui/ViewUtils.java

File: app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/playlist/YoutubeActivity.java [hunk 2]
```diff
@@ -87,7 +92,9 @@ private void setUpListClick() {
         playlistViewModel.getSelected().observe(this, youtubePlaylist -> {
             if (youtubePlaylist != null) {
                 Toast.makeText(YoutubeActivity.this, "you selected a " + youtubePlaylist.getPlaylistId(), Toast.LENGTH_SHORT).show();
-                //TODO: Transfer Intent
+                Intent intent = new Intent(this, VideoActivity.class);
+                intent.putExtra("playlistId", youtubePlaylist.getPlaylistId());
```
Comment (dr0pdb) [inline app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/playlist/YoutubeActivity.java]: Define the string as a public static const variable and use it.

File: app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/video/VideoViewModel.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+import javax.inject.Inject;
+
+public class VideoViewModel extends ViewModel {
+    @VisibleForTesting
+    static final String VIDEO_VIEW_MODEL = "videoViewModel";
+
+    private final RateLimiter<String> repoRateListLimit;
+    public ObservableInt loading;
+    public ObservableInt showEmpty;
```
Comment (dr0pdb) [inline app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/video/VideoViewModel.java:22]: Any way we could use `SingleEventLiveData`?

**Review by dr0pdb — CHANGES_REQUESTED** [review]

**Review by ravi-kishan — APPROVED** [review]

File: app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/video/VideoViewModel.java (hunk 3 above)
Comment (dr0pdb) [reply app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/video/VideoViewModel.java:22]: @sheetal-g9 ?

Comment (sheetal-g9) [reply app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/video/VideoViewModel.java:22]: I am using this just to observe the visibility of a layout which is of primitive type int.
ObservableInt (which is a subclass of databinding.BaseObservable )would be more suitable here.

Comment (dr0pdb) [reply app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/video/VideoViewModel.java:22]: The use of `showEmpty` is just to signal the Activity that you should show an empty view layout. That's the exact purpose of a `SingleEventLiveData`. It is used to send a one-time signal to the view about something.

**Review by dr0pdb — APPROVED** [review]

**Pull request merged**